modules/meeting/app/components/meetings/delete_dialog_component.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Meetings
+  class DeleteDialogComponent < ApplicationComponent
+    include ApplicationHelper
+    include OpTurbo::Streamable
+
+    def initialize(meeting:, project:)
+      super
+
+      @meeting = meeting
+      @project = project
+    end
+
+    private
+
+    def id = "delete-meeting-dialog"
+
+    def title = I18n.t("meeting.delete_dialog.title")
+
+    def confirmation_message
+      if recurring_meeting.present?
```

Comment:
I would assume this message has three cases:

1. The meeting is a one-off StructuredMeeting (it is neither a `RecurringMeeting` series nor does it have a  `recurring_meeting` assocation).
2. The meeting is part of a meeting series. (It is a `StructuredMeeting`, but has an assocation `recurring_meeting` set. The message would contain something like "This meeting is a part of the series XYZ. This will only delete this occurrence"
4. It is a meeting series object itself (it is a `RecurringMeeting`). "Deleting XYZ series will delete all occurrences..."

Reply:
As I've modelled it, `Meetings::DialogComponent` will always be called with a `Meeting` or `StructuredMeeting`.